tests/basic/volume_ops_test.go

```diff
@@ -3202,6 +3202,85 @@ var _ = Describe("{FioClonedVolumeFaultInjection}", func() {
 		}
 	})
 })
+var _ = Describe("{VolumePreCheck}", func() {
+	/*
+			https://portworx.atlassian.net/browse/PTX-20557
+			1. Deploy a basic volume on the cluster
+			2. Now we need to test the ha-update of the volume with different cases
+			scenarios :
+		    1. Test the ha-update sources option with an uuid of other nodes on which node is not present
+		    2. Test the ha-update sources option with an ip of nodes
+			3. Test the ha-update sources option with an invalid uuid of the node on which the repl of the volume is present
+		    4. Test the ha-update sources option with a valid uuid of the node on which the repl of the volume is present
+
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("volumeprecheck", "Precheck for volume operations", nil, 0)
+	})
+
+	It("volumeprecheck", func() {
+		log.InfoD("volumeprecheck")
+		stepLog := "Create a volume and perform a precheck for sources options"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			stNodes := node.GetStorageDriverNodes()
+			nodesuuidWithoutReplica := make([]string, 0)
+			nodesIP := make([]string, 0)
+			index := rand.Intn(len(stNodes))
+			selectedNode := &stNodes[index]
+			for _, stNode := range stNodes {
+				if stNode.Name != selectedNode.Name {
+					nodesuuidWithoutReplica = append(nodesuuidWithoutReplica, stNode.Id)
+					nodesIP = append(nodesIP, stNode.Addresses[0])
+				}
+			}
+			id := uuid.New()
+			volName := fmt.Sprintf("volume_%s", id.String()[:8])
+			log.InfoD("Create a volume with a min size on node [%s]", selectedNode.Name)
+			basicVolumeCreate := fmt.Sprintf("volume create --nodes %s %s", selectedNode.Id, volName)
+			_, err := runPxctlCommand(basicVolumeCreate, *selectedNode, nil)
+			log.FailOnError(err, "volume creation failed on the cluster with volume name [%s] on node %s", volName, selectedNode.Name)
+			log.InfoD("Base Volume creation with volume name %s successful", volName)
+
+			log.InfoD("Test the ha-update sources option with a uuid of other nodes on which node is not present")
+			wrongUuidcmd := fmt.Sprintf("v ha-update %s --repl 2 --sources %s", volName, nodesuuidWithoutReplica[rand.Intn(len(nodesuuidWithoutReplica))])
```

Comment:
test with aggregated volumes

Reply:
https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-pxe-byoc/2629/console. -- tested with aggregated volumes , pass log

Reply:
parameteize the aggregation and replication base on number of storage nodes and  create volume using replication and aggregation combined

Reply:
added it ,got the pass logs https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-pxe-byoc/2676/console